File: Tugas Modul 4 (ROS)/bev_workspace/src/bev_package/script/service_client_node.py
import rospy
from bev_package.srv import service, serviceResponse

def magnitude_formula(req):
    magnitude = (req.komponenX ** 2 + req.komponenY ** 2 + req.komponenZ ** 2) ** 0.5
    rospy.loginfo("Nilai magnitudenya adalah %f", magnitude)
    return serviceResponse(magnitude)

if __name__ == '__main__':
    rospy.init_node('service_client')
    rospy.wait_for_service('service')
    try:
        client = rospy.ServiceProxy('service', service)
        req = serviceRequest()
        req.komponenX = 2.0
        req.komponenY = 3.0
        req.komponenZ = 6.0
        res = client(req)
        rospy.loginfo("Nilai magnitudenya adalah %f", res.magnitude)
    except rospy.ServiceException as e:
        rospy.loginfo("Gagal untuk memanggil service: %s", e)

File: Tugas Modul 4 (ROS)/bev_workspace/src/bev_package/src/service_client_node.cpp
#include "ros/ros.h"
#include "bev_package/service.h"

int main(int argc, char **argv)
{
  ros::init(argc, argv, "service_client");

  ros::NodeHandle nh;

  ros::ServiceClient client = nh.serviceClient<bev_package::service>("service");

  bev_package::service::Request req;
  req.komponenX = 2.0;
  req.komponenY = 3.0;
  req.komponenZ = 6.0;

  bev_package::service::Response res;
  if (client.call(req, res))
  {
    ROS_INFO("Nilai magnitudenya adalah %f", res.magnitude);
  }
  else  {
    ROS_ERROR("Gagal untuk memanggil service.");
    return 1;
  }

  return 0;
}
